### PR TITLE
[Xtensa] Fix fp16 conversion

### PR DESCRIPTION
**Pull request by gerekon**

### CODE REVIEW

**Comment by gerekon** [comment]
@andreisfr @sstefan1 PTAL
@kassane FYI

File: llvm/test/CodeGen/Xtensa/fp16.ll [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+define void @foo() nounwind {
+; CHECK:          .literal .LCPI0_1, __gnu_h2f_ieee
+; CHECK:          .literal .LCPI0_3, __gnu_f2h_ieee
+; CHECK-ESP32S2:  .literal .LCPI0_1, __gnu_h2f_ieee
+; CHECK-ESP32S2:  .literal .LCPI0_3, __addsf3
+; CHECK-ESP32S2:  .literal .LCPI0_4, __gnu_f2h_ieee
+
+; CHECK-LABEL: foo:
+; CHECK-ESP32S2-LABEL: foo:
+; CHECK:          l32r a5, .LCPI0_1
```
Comment (gerekon) [inline llvm/test/CodeGen/Xtensa/fp16.ll:21]: In test I just added checks for proper calls to conversion funcs

**Comment by sstefan1** [comment]
I suggest using update_llc_test_checks.py for generating check lines. Otherwise LGTM!

**Comment by gerekon** [comment]
@sstefan1 
> I suggest using update_llc_test_checks.py for generating check lines. Otherwise LGTM!

Actually I used it to generate initial test case. But after that I removed instructions and left only ones related to conversion functions calls as ot is done for [ARM](https://github.com/espressif/llvm-project/blob/xtensa_release_17.0.1/llvm/test/CodeGen/ARM/fp16.ll#L29). But looks like that code is 9 years old, so it was written before the first version of `update_llc_test_checks.py` was implemented.

So. Ok. I will re-generate checks for easier maintenance in future.

**Comment by kassane** [comment]
This change didn't give me the error mentioned in the issue.

But, now:
```bash
$> ./out/zig-x86_64-linux-musl-baseline/zig cc -target xtensa-freestanding -mcpu=esp32 start.c
LLVM Emit Object... error: <unknown>:0: Undefined temporary symbol .LBB385_2
```

**Comment by kassane** [comment]
@gerekon, llvm 19.1.2 branch - error f16 returned, re-add your patch.
https://github.com/espressif/llvm-project/issues/91

**Comment by gerekon** [comment]
@kassane Hmm, it is there e17dfae23576

**Comment by gerekon** [comment]
sorry. Look slike the branch on GH is not fully synced

**Comment by gerekon** [comment]
Done. Fixed sync process.

**Pull request merged**